browser/ui/webui/settings/brave_default_extensions_handler.cc

```diff
@@ -509,21 +519,79 @@ void BraveDefaultExtensionsHandler::GetDecentralizedDnsResolveMethodList(
 }
 
 #if BUILDFLAG(IPFS_ENABLED)
-void BraveDefaultExtensionsHandler::OnIpfsLaunched(bool result, int64_t pid) {
-  if (!IsJavascriptAllowed())
+void BraveDefaultExtensionsHandler::FileSelected(const base::FilePath& path,
+                                                 int index,
+                                                 void* params) {
+  ipfs::IpfsService* service =
+      ipfs::IpfsServiceFactory::GetForContext(profile_);
+  if (!service)
     return;
+  service->GetIpnsKeysManager()->ImportKey(
+      path, dialog_key_,
+      base::BindOnce(&BraveDefaultExtensionsHandler::OnKeyImported,
+                     base::Unretained(this)));
```

Comment:
@bbondy @spylogsster I was just randomly browsing for `Unretained` pointers - this is an easily reproducible crash. Just start an import of a file and quickly close the settings tab (to 100% reproduce pick up any large file, I've tried with videos).

Basically the first thing I do during reviews is `Ctrl-F` `Unretained` - hope this could help to prevent future crashes

```
Received signal 11 SEGV_MAPERR ffffd859ca3ff9df
#0 0x7f4936df7659 base::debug::CollectStackTrace()
#1 0x7f4936cf63f3 base::debug::StackTrace::StackTrace()
#2 0x7f4936df71b0 base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f49260363c0 (/usr/lib/x86_64-linux-gnu/libpthread-2.31.so+0x153bf)
#4 0x7f4934c79237 content::WebUIMessageHandler::IsJavascriptAllowed()
#5 0x562157e1730f content::WebUIMessageHandler::FireWebUIListener<>()
#6 0x562157e171ed BraveDefaultExtensionsHandler::OnKeyImported()
#7 0x5621577ab147 ipfs::IpnsKeysManager::OnKeyImported()
#8 0x5621577abef9 base::internal::FunctorTraits<>::Invoke<>()
#9 0x5621577abe27 base::internal::Invoker<>::RunOnce()
#10 0x7f49307157ed _ZNO4base12OnceCallbackIFvNSt4__Cr10unique_ptrINS1_12basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEENS1_14default_deleteIS8_EEEEEE3RunESB_
#11 0x7f4930715f4e network::(anonymous namespace)::SimpleURLLoaderImpl::FinishWithResult()
#12 0x7f4930713fea network::(anonymous namespace)::SimpleURLLoaderImpl::OnReceiveResponse()
#13 0x7f493075050a network::mojom::URLLoaderClientStubDispatch::Accept()
#14 0x7f4936694dab mojo::InterfaceEndpointClient::HandleValidatedMessage()
```

Reply:
thanks https://github.com/brave/brave-browser/issues/15961